ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,868 @@
+from ..io import EventSource, HDF5TableWriter, SimTelEventSource
+from ..io.containers import (
+    DL1CameraContainer,
+    EventIndexContainer,
+    ImageParametersContainer,
+    TelEventIndexContainer,
+    SimulatedShowerDistribution,
+    MorphologyContainer,
+)
+
+tables.parameters.NODE_CACHE_SLOTS = 3000  # fixes problem with too many datasets
+
+PROV = Provenance()
+
+# define the version of the DL1 data model written here. This should be updated
+# when necessary:
+# - increase the major number if there is a breaking change to the model
+#   (meaning readers need to update scripts)
+# - increase the minor number if new columns or datasets are added
+# - increase the patch number if there is a small bugfix to the model.
+DL1_DATA_MODEL_VERSION = "v1.0.0"
+
+
+def write_core_provenance(output_filename, obs_id, subarray, writer):
+    """
+    Attaches Core Provenence headers to an output HDF5 file.
+    Right now this is hard-coded for use with the ctapipe-stage1-process tool
+
+    TODO: generalize, and move to ctapipe.io
+
+    Parameters
+    ----------
+    output_filename: str
+        output HDF5 file
+    obs_id: int
+        observation ID
+    subarray:
+        SubarrayDescription to get metadata from
+    writer: HDF5TableWriter
+        output
+    """
+    import uuid
+
+    activity = PROV.current_activity.provenance
+
+    metadata = {
+        "CTA REFERENCE VERSION ": "1",
+        "CTA CONTACT ORGANIZATION": "CTA Consortium",
+        "CTA CONTACT NAME": "K. Kosack",
+        "CTA CONTACT EMAIL": "karl.kosack@cea.fr",
+        "CTA PRODUCT DESCRIPTION": "DL1 Event List",
+        "CTA PRODUCT CREATION_TIME": Time.now().iso,
+        "CTA PRODUCT ID": str(uuid.uuid4()),
+        "CTA PRODUCT DATA CATEGORY": "S",
+        "CTA PRODUCT DATA LEVEL": "DL1",
+        "CTA PRODUCT DATA TYPE": "Event",
+        "CTA PRODUCT DATA ASSOCIATION": "Subarray",
+        "CTA PRODUCT DATA MODEL NAME": "DL1/Event",
+        "CTA PRODUCT DATA MODEL VERSION": DL1_DATA_MODEL_VERSION,
+        "CTA PRODUCT DATA MODEL URL": "",
+        "CTA PRODUCT FORMAT": "hdf5",
+        "CTA PROCESS TYPE": "simulation",
+        "CTA PROCESS SUBTYPE": "",
+        "CTA PROCESS ID": str(obs_id),
+        "CTA ACTIVITY NAME": activity["activity_name"],
+        "CTA ACTIVITY TYPE": "software",
+        "CTA ACTIVITY ID": activity["activity_uuid"],
+        "CTA ACTIVITY START": activity["start"]["time_utc"],
+        "CTA ACTIVITY SOFTWARE NAME": "ctapipe",
+        "CTA ACTIVITY SOFTWARE VERSION": activity["system"]["ctapipe_version"],
+        "CTA INSTRUMENT SITE": "unknown",
+        "CTA INSTRUMENT CLASS": "subarray",
+        "CTA INSTRUMENT TYPE": "",
+        "CTA INSTRUMENT SUBTYPE": "",
+        "CTA INSTRUMENT VERSION": "",
+        "CTA INSTRUMENT ID": str(subarray),
+    }
+
+    for key, value in metadata.items():
+        writer._h5file.root._v_attrs[key] = value
+
+
+def morphology(geom, image_mask) -> MorphologyContainer:
```

Comment:
Is the calculation of number of islands usually known as `Morphology`? It may be, but I do not seem to recall hearing it before...

Reply:
me neither...

But Morphology is probably meant as kind of a catch all container for non-hillas image related parameters, so islands fits here

Reply:
I agree we can imagine more than just number of islands to be added to this "morphology" container at some point. I quite like the name actually!

Reply:
yes, what @MaxNoe said is right - I couldn't think of a better name, and we could add other image-morphology parameters here later. Maybe Topology is better?

Reply:
Topology also came to my mind as well, but I'd say we can change that name if we wish according to its "final" content at a later point...